snuba/datasets/entities/spans.py

```diff
@@ -0,0 +1,88 @@
+from datetime import timedelta
+from typing import Mapping, Sequence
+
+from snuba.clickhouse.columns import UUID, ColumnSet, DateTime, Nested
+from snuba.clickhouse.columns import SchemaModifiers as Modifiers
+from snuba.clickhouse.columns import String, UInt
+from snuba.clickhouse.translators.snuba.mappers import SubscriptableMapper
+from snuba.clickhouse.translators.snuba.mapping import TranslationMappers
+from snuba.datasets.entities import EntityKey
+from snuba.datasets.entity import Entity
+from snuba.datasets.plans.single_storage import SingleStorageQueryPlanBuilder
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_writable_storage
+from snuba.pipeline.simple_pipeline import SimplePipelineBuilder
+from snuba.query.data_source.join import ColumnEquivalence, JoinRelationship, JoinType
+from snuba.query.extensions import QueryExtension
+from snuba.query.processors import QueryProcessor
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+from snuba.query.project_extension import ProjectExtension
+from snuba.query.timeseries_extension import TimeSeriesExtension
+
+
+class SpansEntity(Entity):
+    def __init__(self) -> None:
+        storage = get_writable_storage(StorageKey.SPANS)
+
+        super().__init__(
+            storages=[storage],
+            query_pipeline_builder=SimplePipelineBuilder(
+                query_plan_builder=SingleStorageQueryPlanBuilder(
+                    storage=storage,
+                    mappers=TranslationMappers(
+                        subscriptables=[
+                            SubscriptableMapper(None, "tags", None, "tags")
+                        ],
+                    ),
+                ),
+            ),
+            abstract_column_set=ColumnSet(
+                [
+                    ("project_id", UInt(64)),
+                    ("transaction_id", UUID()),
+                    ("trace_id", UUID()),
+                    ("transaction_span_id", UInt(64)),
+                    ("span_id", UInt(64)),
+                    ("parent_span_id", UInt(64, Modifiers(nullable=True))),
+                    ("transaction_name", String()),
+                    ("op", String()),
+                    ("status", UInt(8)),
+                    ("start_ts", DateTime()),
+                    ("start_ns", UInt(32)),
+                    ("finish_ts", DateTime()),
+                    ("finish_ns", UInt(32)),
+                    ("duration_ms", UInt(32)),
+                    ("tags", Nested([("key", String()), ("value", String())])),
+                ]
+            ),
+            join_relationships={
+                "contained": JoinRelationship(
```

Comment:
@evanh  I think we might want to revisit this data structure. Since the name of the relationship is the key of the dictionary, we cannot have two entities with the same name but different right side entity.
This seems a limitation that does not buy us anything (we can always disambiguate relationships by the right side) and make it hard to keep a data model at the same time future proof and intuitive.
`contains` is a fairly common word so we could expect there will be the need for another `contains` relationship in the future, it seems reasonably appropriate for transactions-spans relationship and it keeps the model intuitive. I could find another word that is more specific to this relationship but that would make the model less intuitive for no reason.

Reply:
I sort of thought we would do something like `contains-spans` to differentiate different `contains` relationships. I suppose we could lookup the relationship using rhs entity name as well. It doesn't seem difficult to have the product just not use the same keys twice.

Reply:
> It doesn't seem difficult to have the product just not use the same keys twice.

I am not sure about this, thus my future proofing point. The need to reuse a word will come up when we will add a new entity that was unforeseen before (now there is only one containment relationship in transactions, so the name is obvious, but what happens when we add another entity that is contained by a transaction ? This is very hard to predict). At that point we would have to append something to the new relationship name, since we cannot easily change a name that is already in use. Thus we will remain with inconsistent naming.

If am ok keeping the naming unique  `contains-spans` such that there is no ambiguity aside for the fact that it seems more verbose than it needs, though at that point can we remove the destination entity name from the language? That information will be redundant. `(s: Spans) -[contained_in_transaction]-> t`

What do you think ?